src/convert_image/x86.rs

```diff
@@ -644,140 +1001,12 @@ pub fn rgb_lrgb_bgra_lrgb(
-            for _ in (0..multi_swap_iterations).step_by(ITEMS_PER_ITERATION) {
-                *(obuffer.add(obuffer_offset) as *mut i64) = _bswap64(
-                    (((*(ibuffer.add(ibuffer_offset) as *const u64) >> SHIFT_16) & LOW_MASK)
-                        | ((*(ibuffer.add(ibuffer_offset) as *const u64) << SHIFT_40) & HIGH_MASK)
-                        | ALPHAS_MASK) as i64,
-                );
-
-                *(obuffer.add(obuffer_offset + 8) as *mut i64) = _bswap64(
-                    (((*(ibuffer.add(ibuffer_offset + 6) as *const u64) >> SHIFT_16) & LOW_MASK)
-                        | ((*(ibuffer.add(ibuffer_offset + 6) as *const u64) << SHIFT_40)
-                            & HIGH_MASK)
-                        | ALPHAS_MASK) as i64,
-                );
-
-                *(obuffer.add(obuffer_offset + 16) as *mut i64) = _bswap64(
-                    (((*(ibuffer.add(ibuffer_offset + 12) as *const u64) >> SHIFT_16) & LOW_MASK)
-                        | ((*(ibuffer.add(ibuffer_offset + 12) as *const u64) << SHIFT_40)
-                            & HIGH_MASK)
-                        | ALPHAS_MASK) as i64,
-                );
-
-                *(obuffer.add(obuffer_offset + 24) as *mut i64) = _bswap64(
-                    (((*(ibuffer.add(ibuffer_offset + 18) as *const u64) >> SHIFT_16) & LOW_MASK)
-                        | ((*(ibuffer.add(ibuffer_offset + 18) as *const u64) << SHIFT_40)
-                            & HIGH_MASK)
-                        | ALPHAS_MASK) as i64,
-                );
-
-                x += ITEMS_PER_ITERATION;
-                ibuffer_offset += INPUT_BPP * ITEMS_PER_ITERATION;
-                obuffer_offset += OUTPUT_BPP * ITEMS_PER_ITERATION;
-            }
-
-            // Retrieves the ramaining colors in the line
-            for _ in x..single_swap_iterations {
-                *(obuffer.add(obuffer_offset) as *mut i32) = _bswap(
-                    ((*(ibuffer.add(ibuffer_offset) as *const u32) << SHIFT_8) | 0xFF) as i32,
-                );
-
-                x += 1;
-                ibuffer_offset += INPUT_BPP;
-                obuffer_offset += OUTPUT_BPP;
-            }
-
-            // If the input stride is not at least 1 byte it directly copies byte per byte,
-            // this could happen once per line
-            if x < w {
-                *obuffer.add(obuffer_offset + 0) = *ibuffer.add(ibuffer_offset + 2);
-                *obuffer.add(obuffer_offset + 1) = *ibuffer.add(ibuffer_offset + 1);
-                *obuffer.add(obuffer_offset + 2) = *ibuffer.add(ibuffer_offset + 0);
-                *obuffer.add(obuffer_offset + 3) = 0xFF;
-
-                ibuffer_offset += INPUT_BPP;
-                obuffer_offset += OUTPUT_BPP;
-            }
-
-            ibuffer_offset += input_stride_diff;
-            obuffer_offset += output_stride_diff;
-        }
-    }
-    return true;
+    rgb_to_bgra(
```

Comment:
I think this pull request addresses the addition of i420, so please file this factor-out work in a separate pull-request.
This guarantees the pull is a minimal one.